### DISCUSSION TITLE
Registering Schema with Model Server

### DISCUSSION THREAD

**Discussion by GaryKopp** [eclipse-emfcloud/emfcloud#95] [ModelServer] [answered]
In the coffee editor, the type schema, the meta-model in my way of looking at things, is registered with the Model Server in https://github.com/eclipse-emfcloud/coffee-editor/blob/ab4835d45ddcd9d16af960be19c96e34abff5c47/backend/plugins/org.eclipse.emfcloud.coffee.modelserver.app/src/org/eclipse/emfcloud/coffee/modelserver/app/Application.java, using a call to addEPackageConfigurations (now deprecated; I've forgotten what it is replaced with, but it seems to have similar functionality). In ecore-glsp there is no such registration call. Instead ecore-glsp seems to dynamically access the type schema as ecore.ecore whenever it needs it. Why doesn't ecore-glsp register the type schema's .class files with Model Server in the same way as the coffee editor? I'm not sure what "registration" does (can someone point me to an explanation?). If such registration was added to ecore-glsp would it eliminate a lot of code dealing with accessing the type schema? Or just completely screw things up? :-)

**Comment by agonzalezgaliana** [comment]
I would be very interested to know about the answer to this question. Were you able to find and answer to it?

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi,

The Ecore GLSP Model Server is packaged as a headless Eclipse Application. As such, it can leverage Eclipse extension points. EPackages are registered via an extension point in one of the plugin.xml files.

Ecore.ecore is registered directly by the EMF plug-in
enotation.ecore is registered by org.eclipse.emfcloud.ecore.modelserver: https://github.com/eclipse-emfcloud/ecore-glsp/blob/master/server/org.eclipse.emfcloud.ecore.modelserver/plugin.xml
